src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -35,19 +39,17 @@
     /**
      * Constructor.
      *
-     * @param string        $name   The name of the node
-     * @param NodeInterface $parent The parent of this node
+     * @param string        $name          The name of the node
+     * @param NodeInterface $parent        The parent of this node
+     * @param string        $pathSeparator The Path Separator that is used
      *
-     * @throws \InvalidArgumentException if the name contains a period.
+     * @throws \InvalidArgumentException if the name contains path separator.
      */
-    public function __construct($name, NodeInterface $parent = null)
+    public function __construct($name, NodeInterface $parent = null, $pathSeparator = self::DEFAULT_PATH_SEPARATOR)
     {
-        if (false !== strpos($name, '.')) {
-            throw new \InvalidArgumentException('The name must not contain ".".');
```

Comment:
Why not, but this is the only place where the name is checked. In prototyped node, name property is not enforced and make Node inconsistent.